### ISSUE TITLE
Use hbase counters implementation instead of PutRequest

### ISSUE THREAD

**Issue by pinifaran** [OpenTSDB/opentsdb#677]
We are using open TSDB for storing counters, so that in some flows the same counter (metric name + tags + timestamp) can be updated. Using hbase counters (instead of query and put request) provides a more suitable implementation, and is considered to have better performance in general, so it can be used as the underlying implementation for all writes to open TSDB.

I already have implementation code over TSDB that seems to work ok. Also note that hbase counters can be also stored via asynch hbase (as put request).

**Comment by manolama** [comment]
Yeah we definitely want to enable counter support for such situations but we don't want to have counters for everything in the TSD as it's actually more expensive than a straight put (it's similar to an append in HBase in that the region server has to perform the lock, read, increment, unlock, return code and pushes the returned byte array back to HBase. Puts though, just write a new column in the wal and store file and the response returned is a null unless there's an exception.
But yeah, atomic increments are definitely better than doing a read, update, put (actually you'd want a CAS call). I'll give your PR a review soon and thanks!

**Comment by pinifaran** [comment]
Great, thanks! 
I've created a branch called "V2.1.0 hbase counters" with the implementation I added for using HBase counters. It would be really great if it can be added to a future open TSDB release (configurable, of course).

**Comment by manolama** [comment]
Yeah we'll definitely get it in there, probably with 2.4, maybe 2.5
